src/ctx.cpp

```diff
@@ -2903,6 +2906,7 @@ llvm::Value *FunctionEmitContext::gather(llvm::Value *ptr, const PointerType *pt
 
     // bool type is stored as i8. So, it requires some processing.
     if (returnType->IsBoolType()) {
+        // TODO!: should we just call SwitchBoolToStorageType here unconditionally?
```

Comment:
Yeah, I think this logic should be consolidated in `SwitchBoolToMaskType`. If you can spend a little more time on it, it would be good to 
1. resolve this TODO
2. describe all the bool/mask conversion behavior in design document